File: Backend/Models/ReporteVenta.cs
using System.ComponentModel.DataAnnotations;

public class ReporteVenta
{
  [Key]
  public int ID_Reporte { get; set; }
  public int ID_Agente { get; set; } // Asegúrate de que el tipo sea entero
  public string DescripcionArticulo { get; set; }
  public DateTime FechaVenta { get; set; }
  public decimal ImporteTotal { get; set; }
  public string EstatusVenta { get; set; }
  public string EstatusSupervision { get; set; }
}

File: Mobile/lib/models/reporte_venta.dart
class ReporteVenta {
  final int idReporte;
  final int idAgente;
  final String nombreAgente;
  final String descripcionArticulo;
  final DateTime fechaVenta;
  final double importeTotal;
  final String estatusVenta;
  final String estatusSupervision;

  ReporteVenta({
    required this.idReporte,
    required this.idAgente,
    required this.nombreAgente,
    required this.descripcionArticulo,
    required this.fechaVenta,
    required this.importeTotal,
    required this.estatusVenta,
    required this.estatusSupervision,
  });

  factory ReporteVenta.fromJson(Map<String, dynamic> json) {
    return ReporteVenta(
      idReporte: json['idReporte'] ?? 0,
      idAgente: json['idAgente'] ?? 0,
      nombreAgente: json['nombreAgente'] ?? '',
      descripcionArticulo: json['descripcionArticulo'] ?? '',
      fechaVenta: DateTime.parse(json['fechaVenta']),
      importeTotal: (json['importeTotal'] ?? 0).toDouble(),
      estatusVenta: json['estatusVenta'] ?? '',
      estatusSupervision: json['estatusSupervision'] ?? '',
    );
  }
}
